trie/resolver.go

```diff
@@ -224,7 +225,9 @@ func (tr *TrieResolver) Walker(keyIdx int, k []byte, v []byte) (bool, error) {
 			}
 			encodeLen := tr.a.EncodingLengthForHashing()
 			if int(encodeLen) > len(tr.buf) {
-				tr.buf = make([]byte, encodeLen)
+				pool.PutBuffer(pool.NewValue(tr.buf))
+				buf := pool.GetBuffer(encodeLen)
+				tr.buf = buf.B
```

Comment:
You can now remove `buf` field from the type `Resolver` and just use the buffer from the pool?

Reply:
It'd be nice to do. Checking...

Reply:
And it works :)